server/src/test/java/org/elasticsearch/ingest/IngestServiceTests.java

```diff
@@ -2680,7 +2680,19 @@ private static void executeFailingDelete(DeletePipelineRequest request, ClusterS
     }
 
     private static List<IngestService.PipelineClusterStateUpdateTask> oneTask(PutPipelineRequest request) {
-        return List.of(new IngestService.PutPipelineClusterStateUpdateTask(ActionListener.running(() -> fail("not called")), request));
+        return List.of(new IngestService.PutPipelineClusterStateUpdateTask(createTestListener(), request));
+    }
+
+    private static <T> ActionListener<T> createTestListener() {
+        return new ActionListener<>() {
+            @Override
+            public void onResponse(T t) {}
+
+            @Override
+            public void onFailure(Exception e) {
+                throw new AssertionError(e);
+            }
```

Comment:
Suggest `org.elasticsearch.action.support.ActionTestUtils#assertNoFailureListener` for all these listeners that assert they don't fail.